### PR TITLE
feat(legacy): CEXT-3832 - remove legacy mesh references

### PR DESCRIPTION
**Pull request by amolina-adobe**
## Description
Removes all references to the legacy (MGQL) mesh from the CLI plugin command output.

### Affects the following commands:
* create
* describe
* status
* delete

### Includes:
1. removal the legacy mesh url
2. removal of legacy mesh statuses
3. removal of any migration notices
4. removal of api key creation, subscription, and deletion
5. re-labeling of "Edge mesh" to "Mesh" as it is the sole output
6. minor updates to various command messaging

## Related Issue
https://jira.corp.adobe.com/browse/CEXT-3832

## Motivation and Context
Coerce customers to migrate to edge mesh in production, due to complete deprecation of the legacy mesh (MGQL). 

## How Has This Been Tested?
* Verified unit tests pass
* Verified CLI tests pass after updating them to account for new messaging
* Manual testing of all commands
* Verified absence or errors in Dev Console UI when creating a mesh with new plugin (without an API Key)
* Verified rolling back to an old version of the plugin still works with a mesh created the new plugin (without an API Key)

## Screenshots (if appropriate):
image

#### `aio api-mesh create`
image

#### `aio api-mesh describe`
image

#### `aio api-mesh get` (for good measure)
image

#### `aio api-mesh status`
image

image

#### `aio api-mesh update` (for good measure)
image

#### `aio api-mesh delete`
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/api-mesh/status.js [hunk 1]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
+
+		try {
+			const mesh = await getMesh(imsOrgId, projectId, workspaceId, workspaceName, meshId);
+			this.log(''.padEnd(102, '*'));
+			await this.displayMeshStatus(mesh, imsOrgCode, projectId, workspaceId);
+			this.log(''.padEnd(102, '*'));
+		} catch (err) {
+			// Error occurred while fetching/displaying the mesh status
+			this.error(
+				`Unable to get the mesh status. If the error persists please contact support. RequestId: ${global.requestId}`,
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:52]:
```suggestion
				`Unable to get mesh status. If this error persists, contact support. RequestId: ${global.requestId}`,
```

File: src/commands/api-mesh/status.js [hunk 2]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
-				this.log(
-					meshLabel === 'Your mesh'
-						? `${meshLabel} errored out with the following error.`
-						: `${meshLabel} build has errors.`,
-				);
-				this.log(mesh.error);
+				this.log('Mesh build has errors.', mesh.error);
+				break;
+			default:
+				this.log('Mesh status is not available. Please wait for a while and try again.');
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:93]:
```suggestion
				this.log('Mesh status is not available. Wait a few minutes and try again.');
```

File: src/commands/api-mesh/status.js [hunk 3]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
+			workspaceId,
+			mesh.meshId,
+		);
+
+		const meshDeploymentStatus = String(meshDeployments.status).toLowerCase();
+
+		switch (meshDeploymentStatus) {
+			case 'provisioning':
+				this.log(
+					`Mesh is currently being provisioned. Please wait a few minutes before checking again.`,
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:120]:
```suggestion
					`Currently provisioning your mesh. Wait a few minutes and check again.`,
```

File: src/commands/api-mesh/status.js [hunk 4]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
+				);
+				break;
+			case 'de-provisioning':
+				this.log(
+					`Mesh is currently being de-provisioned. Please wait a few minutes before checking again.`,
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:125]:
```suggestion
					`Currently de-provisioning your mesh. Wait a few minutes and check again.`,
```

File: src/commands/api-mesh/status.js [hunk 5]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
+				);
+				break;
+			case 'success':
+				this.log(`Mesh has been successfully built.`);
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:129]:
```suggestion
				this.log(`Mesh was built successfully.`);
```

File: src/commands/api-mesh/status.js [hunk 6]
```diff
@@ -28,123 +27,113 @@ class StatusCommand extends Command {
...
+				break;
+			case 'error':
+				this.log('Mesh build has errors.', meshDeployments.error);
+				break;
+			default:
+				this.log('Mesh status is not available. Please wait for a while and try again.');
```
Comment (jhadobe) [inline src/commands/api-mesh/status.js:135]:
```suggestion
				this.log('Mesh status is not available. Wait a few minutes and try again.');
```

**Review by jhadobe — CHANGES_REQUESTED** [review]

File: src/lib/devConsole.js [hunk 7]
```diff
@@ -1000,9 +952,9 @@ const getMeshDeployments = async (organizationCode, projectId, workspaceId, mesh
 		logger.error(`Error fetching deployments for mesh: ${meshId}`);
 
 		return {
-			status: 'ERROR',
```
Comment (amolina-adobe) [inline src/lib/devConsole.js:1003]: The `error` status has special meaning in the deployments table. It signifies that an error occurred during provisioning. 

However, an error here signifies that we were unable to fetch the deployment status altogether. Thus, we don't want to default to that status here. Changed to `null` instead

**Review by revanth0212 — DISMISSED** [review]

**Review by kmaschi — APPROVED** [review]

**Review by kmaschi — APPROVED** [review]

**Review by revanth0212 — APPROVED** [review]

**Pull request merged**